ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -88,17 +88,15 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, NetworkMetr
    * @param pollTimeoutMs the poll timeout.
    * @return a list of {@link ResponseInfo} representing the responses received for any requests that were sent out
    * so far.
-   * @throws IOException if the {@link Selector} associated with this NetworkClient throws
```

Comment:
I did think about this before making the change. 
The nio `Selector` throws an `IOException` if the `select()` throws an `IOException`. It doesn't say anywhere that this error is irrecoverable or that it is fatal - I think we only discussed about not knowing why an `IOException` would be received. It is wasteful to bail out completely if at all this happens.
Secondly, I thought about whether to propagate it all the way to the router. It seemed like the handling should be done here because the `Selector` is a `NetworkClient` thing and I like the abstraction of the `NetworkClient` taking a request and either sending it successfully, failing it or timing it out.